topi/include/topi/image/resize.h

```diff
@@ -0,0 +1,288 @@
+
+namespace topi {
+namespace image {
+using namespace tvm;
+
+/*!
+* \brief Resize given tensor to given shape using nearest neighbour for NHWC
+*
+* \param inputs The input tensor array.
+* \param shape Output shape to resize to.
+* \param align_corners To preserve centers of 4 corner pixels
+* \param name Name of the operation
+* \param tag The tag to mark the operation
+*
+* \return A Tensor resized to given shape
+*/
+inline Tensor resize_nn_nhwc(const Array<Tensor>& inputs,
+                             const Array<Expr>& shape,
+                             bool align_corners = false,
+                             std::string name = "tensor",
+                             std::string tag = kInjective) {
+  Array<Expr> out_shape;
+  out_shape.push_back(inputs[0]->shape[0]);
+  out_shape.push_back(shape[0]);
+  out_shape.push_back(shape[1]);
+  out_shape.push_back(inputs[0]->shape[3]);
+
+  Expr h_ratio = shape[0] / inputs[0]->shape[1];
+  Expr w_ratio = shape[1] / inputs[0]->shape[2];
+
+  return compute(
+    out_shape, [&](const Array<Var>& indices) {
+    Array<Expr> idx;
+    idx.push_back(indices[0]);
+    idx.push_back(indices[1] / h_ratio);
+    idx.push_back(indices[2] / w_ratio);
+    idx.push_back(indices[3]);
+
+    return inputs[0](idx);
+    }, name, tag);
+}
+
+/*!
+* \brief Resize given tensor to given shape using nearest neighbour for NCHW
+*
+* \param inputs The input tensor array.
+* \param shape Output shape to resize to.
+* \param align_corners To preserve centers of 4 corner pixels
+* \param name Name of the operation
+* \param tag The tag to mark the operation
+*
+* \return A Tensor resized to given shape
+*/
+inline Tensor resize_nn_nchw(const Array<Tensor>& inputs,
+                             const Array<Expr>& shape,
+                             bool align_corners = false,
+                             std::string name = "tensor",
+                             std::string tag = kInjective) {
+  Array<Expr> out_shape;
+  out_shape.push_back(inputs[0]->shape[0]);
+  out_shape.push_back(inputs[0]->shape[1]);
+  out_shape.push_back(shape[0]);
+  out_shape.push_back(shape[1]);
+
+  Expr h_ratio = shape[0] / inputs[0]->shape[2];
+  Expr w_ratio = shape[1] / inputs[0]->shape[3];
+
+  return compute(
+    out_shape, [&](const Array<Var>& indices) {
+    Array<Expr> idx;
+    idx.push_back(indices[0]);
+    idx.push_back(indices[1]);
+    idx.push_back(indices[2] / h_ratio);
+    idx.push_back(indices[3] / w_ratio);
+
+    return inputs[0](idx);
+    }, name, tag);
+}
+
+/*!
+* \brief Resize given tensor to given shape using nearest neighbour
+*
+* \param inputs The input tensor array.
+* \param shape Output shape to resize to.
+* \param layout input layout
+* \param align_corners To preserve centers of 4 corner pixels
+* \param name Name of the operation
+* \param tag The tag to mark the operation
+*
+* \return A Tensor resized to given shape
+*/
+inline Tensor resize_nn(const Array<Tensor>& inputs,
```

Comment:
what does nn mean? if it is nearest neighbor, consider use a full name, nn usually means "neural network" in the first context

Reply:
Thanks. Changed to nearest_neighbor.